pkg/lvgl/contrib/lvgl.c

```diff
@@ -104,21 +112,22 @@ void lvgl_init(screen_dev_t *screen_dev)
     _screen_dev = screen_dev;
     assert(screen_dev->display);
 
-    lv_disp_drv_t disp_drv;
+    lv_disp_draw_buf_init(&disp_buf, draw_buf, NULL, LVGL_COLOR_BUF_SIZE);
+
+    static lv_disp_drv_t disp_drv;
```

Comment:
this avoids a hardfault and I was stuck because of that until @kaspar030 found the solution :)

Reply:
Yes but I find it weird to have it inside the function, `disp_buf` is also only used inside the function scope, do you explicitly want `disp_drv` to only be visible inside the function? For example in #16944 I needed a pointer to that driver, and this would prevent me from accessing it. Anyway just arguing for consistency.

Reply:
just making it static was the quick fix in the upstream examples. it could be file scope, or when namespaced, a global symbol.
I propose changing when needed.

Reply:
(it can't be on stack anymore, in lvgl8, the ptr is copied not that whole struct).

Reply:
That was not my point, anyway, non important

Reply:
I can try to make it file scope, should not change too much and if it can make things simpler for you, why not

Reply:
This one is already green, I'll move it If I have to, that PR still needs some work.

Reply:
I can do it here, I find it cleaner to declare it at file scope and there are other Kconfig stuff that I'd like to change again.